### ISSUE TITLE
Flakboy 2: the list of levels is not showing 

### ISSUE THREAD

**Issue by Maxwelllink** [ruffle-rs/ruffle#12370]
### Describe the bug

The list of levels is not showing after selecting a game mode. Some buttons do not work (eg language selection). Game mode selection not showing after restart
Game: https://archive.org/details/flakboy-flash-game-series.

https://github.com/ruffle-rs/ruffle/assets/67246937/b2f05d63-dd21-4386-93fd-77d88829e50a

### Expected behavior

Level selection and language selection should work correctly

### Affected platform

Online demo

### Operating system

Android 13

### Browser

Kiwi Browser, Google Chrome

**Comment by Lord-McSweeney** [comment]
Can you ZIP and upload the SWF to github, please?

**Comment by n0samu** [comment]
This seems to be the correct link to the SWF: https://archive.org/download/flakboy-flash-game-series./Flakboy2_v2.swf

I think this is a performance issue with your device, not a bug in Ruffle. It takes a few seconds for the level selection screen to appear on my PC, so maybe on your device it is taking longer than 15 seconds and hitting a script timeout.

**Comment by Maxwelllink** [comment]
> This seems to be the correct link to the SWF: https://archive.org/download/flakboy-flash-game-series./Flakboy2_v2.swf
> 
> I think this is a performance issue with your device, not a bug in Ruffle. It takes a few seconds for the level selection screen to appear on my PC, so maybe on your device it is taking longer than 15 seconds and hitting a script timeout.

Yes, it works better on PC. But the problem with the restart remained

https://github.com/ruffle-rs/ruffle/assets/67246937/2692a742-646c-4940-b1b9-0c231766053e

**Comment by Maxwelllink** [comment]
Damn... Miss click, I accidentally closed this

**Comment by n0samu** [comment]
The error when starting the game a second time is:
```
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "click", class: flash.events::MouseEvent, ptr: 0x2112de69700 }) to handler FunctionObject(FunctionObject { ptr: 0x2112dccf020 }) : TypeError: Error #1009: Cannot access a property or method of a null object reference. (accessing field: completed)
        at spill.flakboy.data::Data$/isGameComplete()
        at spill.flakboy.menus::ModeSelectScreen()
        at Main/goGameModeSelectScreen()
        at spill.flakboy.menus::MainScreen/showGameModeSelect()
```
This is probably related to unimplemented SharedObject/AMF serialization features.
